src/uv-common.c

```diff
@@ -33,6 +33,62 @@
 # include <net/if.h> /* if_nametoindex */
 #endif
 
+
+static uv_allocator_t uv__allocator = {
+  malloc,
+  realloc,
+  free,
+};
+
+void* uv__calloc(size_t count, size_t size) {
+  /* Note: the multiplcation is not safe if
+   * the result overflows a 'size_t' */
+  size_t total = count * size;
+  void* m = uv__malloc(total);
+  if (m == NULL)
+    return NULL;
+  return memset(m, 0, total);
+}
+
+char* uv__strdup(const char *s) {
+  size_t len = strlen(s) + 1;
+  char* m = uv__malloc(len);
+  if (m == NULL)
+    return NULL;
+  return memcpy(m, s, len);
+}
+
+char* uv__strndup(const char *s, size_t n) {
+  size_t len = strnlen(s, n);
+  char* m = uv__malloc(len + 1);
+  if (m == NULL)
+    return NULL;
+  m[len] = '\0';
+  return memcpy(m, s, len);
+}
+
+void* uv__malloc(size_t size) {
+  return uv__allocator.local_malloc(size);
+}
+
+void uv__free(void* ptr) {
+  uv__allocator.local_free(ptr);
+}
+
+void* uv__realloc(void* ptr, size_t size) {
+  return uv__allocator.local_realloc(ptr, size);
+}
+
+int uv_replace_allocator(const uv_allocator_t *allocator) {
+  if (!allocator->local_malloc ||
+      !allocator->local_realloc ||
+      !allocator->local_free)
```

Comment:
Can you check `== NULL` instead of doing boolean coercion?  Also, maybe add braces around the body.